pkg/kube/snapshot/snapshot_test.go

```diff
@@ -1458,7 +1607,83 @@ func (s *SnapshotLocalTestSuite) TestLabels(c *C) {
 		}
 	}
 }
-
+func (s *SnapshotLocalTestSuite) TestAnnotations(c *C) {
+	ctx := context.Background()
+	scheme := runtime.NewScheme()
+	scheme.AddKnownTypeWithName(schema.GroupVersionKind{Group: "snapshot.storage.k8s.io", Version: "v1alpha1", Kind: "VolumeSnapshotList"}, &unstructured.UnstructuredList{})
+	scheme.AddKnownTypeWithName(schema.GroupVersionKind{Group: "snapshot.storage.k8s.io", Version: "v1beta1", Kind: "VolumeSnapshotList"}, &unstructured.UnstructuredList{})
+	scheme.AddKnownTypeWithName(schema.GroupVersionKind{Group: "snapshot.storage.k8s.io", Version: "v1", Kind: "VolumeSnapshotList"}, &unstructured.UnstructuredList{})
+	ns := "namespace"
+	volName := "vol1"
+	snapName := "snap1"
+	snapClass := "snapClass"
+	fakeCli := fake.NewSimpleClientset(fakePVC(volName, ns))
+	for _, tc := range []struct {
+		dynCli              dynamic.Interface
+		snapshotAnnotations map[string]string
+		expectedAnnotations map[string]string
+		errChecker          Checker
+	}{
+		{
+			dynCli: dynfake.NewSimpleDynamicClient(scheme),
+			snapshotAnnotations: map[string]string{
+				"annotationtest": "true",
+			},
+			expectedAnnotations: map[string]string{
+				"annotationtest": "true",
+			},
+			errChecker: IsNil,
+		},
+		{ // empty annotations list
+			dynCli:              dynfake.NewSimpleDynamicClient(scheme),
+			snapshotAnnotations: map[string]string{},
+			expectedAnnotations: map[string]string{},
+			errChecker:          IsNil,
+		},
+		{ // annotations list matches
+			dynCli: dynfake.NewSimpleDynamicClient(scheme),
+			snapshotAnnotations: map[string]string{
+				"annotationtest":  "true",
+				"annotationtest1": "false",
+			},
+			expectedAnnotations: map[string]string{
+				"annotationtest":  "true",
+				"annotationtest1": "false",
+			},
+			errChecker: IsNil,
+		},
+		{ // nil lists
+			dynCli:     dynfake.NewSimpleDynamicClient(scheme),
+			errChecker: IsNil,
+		},
+	} {
+		for _, fakeSs := range []snapshot.Snapshotter{
+			snapshot.NewSnapshotAlpha(fakeCli, tc.dynCli),
+			snapshot.NewSnapshotBeta(fakeCli, tc.dynCli),
+			snapshot.NewSnapshotStable(fakeCli, tc.dynCli),
+		} {
+			var err error
+			var vs *snapv1.VolumeSnapshot
+			snapshotMeta := snapshot.ObjectMeta{
+				Name:        snapName,
+				Namespace:   ns,
+				Annotations: tc.snapshotAnnotations,
+			}
+			err = fakeSs.Create(ctx, volName, &snapClass, false, snapshotMeta)
+			if err == nil {
+				vs, err = fakeSs.Get(ctx, snapName, ns)
+				annotation := vs.GetAnnotations()
+				c.Assert(len(annotation), Equals, len(tc.expectedAnnotations))
+				for k := range tc.expectedAnnotations {
+					if _, ok := annotation[k]; !ok {
+						c.Assert(false, Equals, true)
+					}
+				}
```

Comment:
[nit] Can we use `DeepEquals` check?
```suggestion
				c.Assert(annotation, DeepEquals, tc.expectedAnnotations)
```

Reply:
done